north/src/main.rs

```diff
@@ -33,10 +33,27 @@ struct Opt {
     pub debug: bool,
 }
 
+async fn parse_config(path: &PathBuf) -> Result<Config> {
+    toml::from_str(&read_to_string(path).await?).context(format!(
+        "Failed to read configuration file {}",
+        path.display()
+    ))
+}
+
+fn unwrap_or_exit<T>(result: Result<T>) -> T {
```

Comment:
A very common approach is to have a 
```rust
async run() -> Result<i32> {
    // read the config
   // run the runtime...
   // etc
}
```
Within that you can use `?` as usual and the match on the `Result` with the `exit` appears only once.

Reply:
But in this case we need to parse the config to set the logging in main before calling run :(

Reply:
Why? Printing the error from `run` should be done via a direct `eprintln` anyway to be independent from the logger.

Reply:
because the log filter is set via either opt.debug || config.debug. The latter coming from the parsed config.